\hypertarget{classMap}{}\doxysection{Map Class Reference}
\label{classMap}\index{Map@{Map}}


Implements the \char`\"{}map\char`\"{} analysis.  




{\ttfamily \#include $<$map.\+h$>$}

Inheritance diagram for Map\+:\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[height=2.000000cm]{classMap}
\end{center}
\end{figure}
\doxysubsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
\mbox{\Hypertarget{classMap_a0f5ad0fd4563497b4214038cbca8b582}\label{classMap_a0f5ad0fd4563497b4214038cbca8b582}} 
\mbox{\hyperlink{classMap_a0f5ad0fd4563497b4214038cbca8b582}{Map}} ()
\begin{DoxyCompactList}\small\item\em Default \mbox{\hyperlink{classMap}{Map}} constructor. \end{DoxyCompactList}\item 
\mbox{\hyperlink{classMap_ad99dc78131acca18ada5e803d0e2a9e7}{Map}} (\mbox{\hyperlink{structParameters}{Parameters}} \&\mbox{\hyperlink{classAnalysis_a8abd4de209925cb46571d4db97977d32}{parameters}}, bool \mbox{\hyperlink{classAnalysis_a744305d7fed1cc583580c5920f0e39b2}{compare\+\_\+groups}}=true, bool \mbox{\hyperlink{classAnalysis_acb2ede1501279963996f5e42eea88cca}{store\+\_\+groups}}=true, bool \mbox{\hyperlink{classAnalysis_a8b6e589cb74d4c25dcdcd88665a07060}{store\+\_\+sequence}}=true)
\begin{DoxyCompactList}\small\item\em Standard \mbox{\hyperlink{classMap}{Map}} constructor. \end{DoxyCompactList}\item 
void \mbox{\hyperlink{classMap_ad084e065185199c750be50698196d840}{extra\+\_\+setup}} ()
\begin{DoxyCompactList}\small\item\em Extra setup steps. \end{DoxyCompactList}\item 
void \mbox{\hyperlink{classMap_a8fa38748c4e54b0dd974915a7a8711c9}{process\+\_\+marker}} (\mbox{\hyperlink{classMarker}{Marker}} \&marker)
\begin{DoxyCompactList}\small\item\em Process a single marker. \end{DoxyCompactList}\item 
void \mbox{\hyperlink{classMap_ad91fb61d49cef731fd4d230634fd4cf9}{generate\+\_\+output}} ()
\begin{DoxyCompactList}\small\item\em Generate the output file. \end{DoxyCompactList}\end{DoxyCompactItemize}
\doxysubsection*{Private Member Functions}
\begin{DoxyCompactItemize}
\item 
void \mbox{\hyperlink{classMap_a06f7bf0741260e57470e78ff5df36a3e}{load\+\_\+contig\+\_\+lengths}} ()
\begin{DoxyCompactList}\small\item\em \char`\"{}map\char`\"{} output file \end{DoxyCompactList}\item 
void \mbox{\hyperlink{classMap_ad085a3c2043db67b97664e30c68183e7}{load\+\_\+bwa\+\_\+index}} ()
\begin{DoxyCompactList}\small\item\em Load a B\+WA index. \end{DoxyCompactList}\end{DoxyCompactItemize}
\doxysubsection*{Private Attributes}
\begin{DoxyCompactItemize}
\item 
\mbox{\Hypertarget{classMap_af2955dd344f2b7eda17d066bf3369429}\label{classMap_af2955dd344f2b7eda17d066bf3369429}} 
\mbox{\hyperlink{structMapResults}{Map\+Results}} \mbox{\hyperlink{classMap_af2955dd344f2b7eda17d066bf3369429}{results}}
\begin{DoxyCompactList}\small\item\em \mbox{\hyperlink{structDistribResults}{Distrib\+Results}} instance to store markers aligned to the genome and total number of retained markers. \end{DoxyCompactList}\item 
\mbox{\Hypertarget{classMap_aa1a4e7c0f0133483a2732f27bd0ef9da}\label{classMap_aa1a4e7c0f0133483a2732f27bd0ef9da}} 
std\+::unordered\+\_\+map$<$ std\+::string, uint64\+\_\+t $>$ \mbox{\hyperlink{classMap_aa1a4e7c0f0133483a2732f27bd0ef9da}{contig\+\_\+lengths}}
\begin{DoxyCompactList}\small\item\em \mbox{\hyperlink{classMap}{Map}} storing the genome\textquotesingle{}s contig lengths \{contig name -\/$>$ contig length\}. \end{DoxyCompactList}\item 
\mbox{\Hypertarget{classMap_a611c3dddeb87aa313c80b01da79aa43b}\label{classMap_a611c3dddeb87aa313c80b01da79aa43b}} 
uint16\+\_\+t \mbox{\hyperlink{classMap_a611c3dddeb87aa313c80b01da79aa43b}{min\+\_\+individuals}} = 0
\begin{DoxyCompactList}\small\item\em Minimum number of individuals in which a marker is present to retain this marker (based on --min-\/freq) \end{DoxyCompactList}\item 
\mbox{\Hypertarget{classMap_a92d00ad3bf2b0dab816d34fee867a968}\label{classMap_a92d00ad3bf2b0dab816d34fee867a968}} 
bwaidx\+\_\+t $\ast$ \mbox{\hyperlink{classMap_a92d00ad3bf2b0dab816d34fee867a968}{index}}
\begin{DoxyCompactList}\small\item\em Pointer to a bwa index created with bwa\+\_\+idx\+\_\+build() and loaded with bwa\+\_\+idx\+\_\+load() \end{DoxyCompactList}\item 
\mbox{\Hypertarget{classMap_aae23996321aaab154bec75427502137f}\label{classMap_aae23996321aaab154bec75427502137f}} 
mem\+\_\+opt\+\_\+t $\ast$ \mbox{\hyperlink{classMap_aae23996321aaab154bec75427502137f}{bwa\+\_\+mem\+\_\+options}}
\begin{DoxyCompactList}\small\item\em Pointer to B\+WA mem options initialized with mem\+\_\+opt\+\_\+init() \end{DoxyCompactList}\item 
\mbox{\Hypertarget{classMap_ab565c5bae20f2e9fdb48996dfdac1b61}\label{classMap_ab565c5bae20f2e9fdb48996dfdac1b61}} 
mem\+\_\+alnreg\+\_\+v \mbox{\hyperlink{classMap_ab565c5bae20f2e9fdb48996dfdac1b61}{alignment\+\_\+results}}
\begin{DoxyCompactList}\small\item\em Store B\+WA mem alignment results from mem\+\_\+align1() \end{DoxyCompactList}\item 
\mbox{\Hypertarget{classMap_a3bb11a7a917041a6fb68c3a519caa56a}\label{classMap_a3bb11a7a917041a6fb68c3a519caa56a}} 
mem\+\_\+aln\+\_\+t \mbox{\hyperlink{classMap_a3bb11a7a917041a6fb68c3a519caa56a}{best\+\_\+alignment}}
\begin{DoxyCompactList}\small\item\em Store a single B\+WA mem alignment results info obtained from a mem\+\_\+alnreg\+\_\+v object with mem\+\_\+reg2aln() \end{DoxyCompactList}\item 
\mbox{\Hypertarget{classMap_a5ec9ce96607680f7eb08bbcb348af0a4}\label{classMap_a5ec9ce96607680f7eb08bbcb348af0a4}} 
double \mbox{\hyperlink{classMap_a5ec9ce96607680f7eb08bbcb348af0a4}{chi\+\_\+squared}}
\begin{DoxyCompactList}\small\item\em Store result of chi-\/squared test on number of individuals from each group. \end{DoxyCompactList}\item 
\mbox{\Hypertarget{classMap_a2dddd9551d1622ea605f108c4dbf80d9}\label{classMap_a2dddd9551d1622ea605f108c4dbf80d9}} 
int \mbox{\hyperlink{classMap_a2dddd9551d1622ea605f108c4dbf80d9}{best\+\_\+alignments}} \mbox{[}3\mbox{]} \{0, -\/1, 0\}
\begin{DoxyCompactList}\small\item\em Store information about best alignments in a mem\+\_\+alnreg\+\_\+v object\+: \{index, score, count\}. \end{DoxyCompactList}\item 
\mbox{\Hypertarget{classMap_aeaebfd45719852317db9e7e3c7ecfbb5}\label{classMap_aeaebfd45719852317db9e7e3c7ecfbb5}} 
\mbox{\hyperlink{structAlignedMarker}{Aligned\+Marker}} \mbox{\hyperlink{classMap_aeaebfd45719852317db9e7e3c7ecfbb5}{aligned\+\_\+marker}}
\begin{DoxyCompactList}\small\item\em Store information about an aligned marker. \end{DoxyCompactList}\item 
\mbox{\Hypertarget{classMap_a857aa986d71a1f496a9931524e4feffe}\label{classMap_a857aa986d71a1f496a9931524e4feffe}} 
std\+::ofstream {\bfseries output\+\_\+file}
\end{DoxyCompactItemize}
\doxysubsection*{Additional Inherited Members}


\doxysubsection{Detailed Description}
Implements the \char`\"{}map\char`\"{} analysis. 

Align all markers from a markers depth table to an assembly using the \char`\"{}mem\char`\"{} algorithm implemented in the B\+WA library (\href{https://github.com/lh3/bwa}{\texttt{ https\+://github.\+com/lh3/bwa}}). ~\newline
For all aligned markers, compute probability of association with groups and group bias (bias score between -\/1 and 1). ~\newline
After all markers are processed, generate a tabulated output file with columns\+: ~\newline
Contig $\vert$ Position $\vert$ Length $\vert$ Marker\+\_\+id $\vert$ Bias $\vert$ P $\vert$ \mbox{\hyperlink{classSignif}{Signif}} 

Definition at line 80 of file map.\+h.



\doxysubsection{Constructor \& Destructor Documentation}
\mbox{\Hypertarget{classMap_ad99dc78131acca18ada5e803d0e2a9e7}\label{classMap_ad99dc78131acca18ada5e803d0e2a9e7}} 
\index{Map@{Map}!Map@{Map}}
\index{Map@{Map}!Map@{Map}}
\doxysubsubsection{\texorpdfstring{Map()}{Map()}}
{\footnotesize\ttfamily Map\+::\+Map (\begin{DoxyParamCaption}\item[{\mbox{\hyperlink{structParameters}{Parameters}} \&}]{parameters,  }\item[{bool}]{compare\+\_\+groups = {\ttfamily true},  }\item[{bool}]{store\+\_\+groups = {\ttfamily true},  }\item[{bool}]{store\+\_\+sequence = {\ttfamily true} }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [inline]}}



Standard \mbox{\hyperlink{classMap}{Map}} constructor. 


\begin{DoxyParams}{Parameters}
{\em parameters} & \mbox{\hyperlink{structParameters}{Parameters}} object storing the value of all R\+A\+D\+Sex parameters \\
\hline
{\em compare\+\_\+groups} & If true, groups are checked when loading the popmap \\
\hline
{\em store\+\_\+groups} & If true, group info is computed for each marker when parsing the markers table \\
\hline
{\em store\+\_\+sequence} & If true, marker sequences are stored for each marker when parsing the markers table \\
\hline
\end{DoxyParams}


Definition at line 100 of file map.\+h.



\doxysubsection{Member Function Documentation}
\mbox{\Hypertarget{classMap_ad084e065185199c750be50698196d840}\label{classMap_ad084e065185199c750be50698196d840}} 
\index{Map@{Map}!extra\_setup@{extra\_setup}}
\index{extra\_setup@{extra\_setup}!Map@{Map}}
\doxysubsubsection{\texorpdfstring{extra\_setup()}{extra\_setup()}}
{\footnotesize\ttfamily void Map\+::extra\+\_\+setup (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [virtual]}}



Extra setup steps. 

Load B\+WA index (build it if it\textquotesingle{}s missing), initalize B\+WA mem parameters, load contig lengths from the genome file, and initialize output file. 

Reimplemented from \mbox{\hyperlink{classAnalysis_a869888497ba90db335e127f3c13d19b4}{Analysis}}.



Definition at line 27 of file map.\+cpp.

\mbox{\Hypertarget{classMap_ad91fb61d49cef731fd4d230634fd4cf9}\label{classMap_ad91fb61d49cef731fd4d230634fd4cf9}} 
\index{Map@{Map}!generate\_output@{generate\_output}}
\index{generate\_output@{generate\_output}!Map@{Map}}
\doxysubsubsection{\texorpdfstring{generate\_output()}{generate\_output()}}
{\footnotesize\ttfamily void Map\+::generate\+\_\+output (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [virtual]}}



Generate the output file. 

Generate a tabulated output file with columns\+: ~\newline
Contig $\vert$ Position $\vert$ Length $\vert$ Marker\+\_\+id $\vert$ Bias $\vert$ P $\vert$ \mbox{\hyperlink{classSignif}{Signif}} 

Reimplemented from \mbox{\hyperlink{classAnalysis_a23da6dbd599ede2e4973a0313128f428}{Analysis}}.



Definition at line 106 of file map.\+cpp.

\mbox{\Hypertarget{classMap_ad085a3c2043db67b97664e30c68183e7}\label{classMap_ad085a3c2043db67b97664e30c68183e7}} 
\index{Map@{Map}!load\_bwa\_index@{load\_bwa\_index}}
\index{load\_bwa\_index@{load\_bwa\_index}!Map@{Map}}
\doxysubsubsection{\texorpdfstring{load\_bwa\_index()}{load\_bwa\_index()}}
{\footnotesize\ttfamily void Map\+::load\+\_\+bwa\+\_\+index (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [private]}}



Load a B\+WA index. 

Check if B\+WA index files exist and build the index if they are missing, then load the index with bwa\+\_\+index\+\_\+load(). 

Definition at line 162 of file map.\+cpp.



Referenced by extra\+\_\+setup().

\mbox{\Hypertarget{classMap_a06f7bf0741260e57470e78ff5df36a3e}\label{classMap_a06f7bf0741260e57470e78ff5df36a3e}} 
\index{Map@{Map}!load\_contig\_lengths@{load\_contig\_lengths}}
\index{load\_contig\_lengths@{load\_contig\_lengths}!Map@{Map}}
\doxysubsubsection{\texorpdfstring{load\_contig\_lengths()}{load\_contig\_lengths()}}
{\footnotesize\ttfamily void Map\+::load\+\_\+contig\+\_\+lengths (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [private]}}



\char`\"{}map\char`\"{} output file 

Load contig lengths

Compute contig lengths from a genome file. 

Definition at line 128 of file map.\+cpp.



Referenced by extra\+\_\+setup().

\mbox{\Hypertarget{classMap_a8fa38748c4e54b0dd974915a7a8711c9}\label{classMap_a8fa38748c4e54b0dd974915a7a8711c9}} 
\index{Map@{Map}!process\_marker@{process\_marker}}
\index{process\_marker@{process\_marker}!Map@{Map}}
\doxysubsubsection{\texorpdfstring{process\_marker()}{process\_marker()}}
{\footnotesize\ttfamily void Map\+::process\+\_\+marker (\begin{DoxyParamCaption}\item[{\mbox{\hyperlink{classMarker}{Marker}} \&}]{marker }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [virtual]}}



Process a single marker. 

Align a marker to the genome and compute probability of association with group and group bias if the marker was aligned.


\begin{DoxyParams}{Parameters}
{\em marker} & Current marker info stored in a \mbox{\hyperlink{classMarker}{Marker}} instance. \\
\hline
\end{DoxyParams}


Reimplemented from \mbox{\hyperlink{classAnalysis_ac2c749dc8512a6ef200042e59c4369bc}{Analysis}}.



Definition at line 51 of file map.\+cpp.



The documentation for this class was generated from the following files\+:\begin{DoxyCompactItemize}
\item 
/home/romain/work/code/\+Rad\+Sex/src/\mbox{\hyperlink{map_8h}{map.\+h}}\item 
/home/romain/work/code/\+Rad\+Sex/src/\mbox{\hyperlink{map_8cpp}{map.\+cpp}}\end{DoxyCompactItemize}
