.. _full-usage:

RADSex usage
============

The RADSex software implements the general command-line interface:

``radsex <command> [options]``

**Available commands**

=======  ===========
Command  Description
=======  ===========
depth    Compute marker depth statistics for each individual
distrib  Compute the distribution of markers between compared groups
freq     Compute the distribution of markers in all individuals
map      Align markers to a genome and compute metrics for each aligned marker
process  Compute a table of marker depths from a set of demultiplexed reads files
signif   Extract markers significantly associated with phenotypic group from a marker depths table
subset   Extract a subset of markers from a marker depths table
=======  ===========


.. _process-usage:

process
-------

Generate a table containing the depth of each marker in each individual from the dataset.

**Command**

::

    radsex process --input-dir input_dir_path --output-file output_file_path [ --threads n_threads --min-depth min_depth ]


**Options**

=====================   ===========
Option                  Description
=====================   ===========
``--input-dir, -i``     Path to a folder containing demultiplexed reads
``--output-file, -o``   Path to the generated marker depths table
``--threads, -T``       Number of threads to use (default: 1)
``--min-depth, -d``     Minimum depth to consider a marker present in an individual (default: 1)
=====================   ===========

**Output**

A :ref:`markers-depths-table-file`:

::

    #Number of markers : 4
    id   sequence     individual_1    individual_2    individual_3    individual_4    individual_5
     0   TGCA..TATT              0              15              24              17              21
     1   TGCA..GACC             20              18               3              26               4
     2   TGCA..ATCG              2               1               5              16               0
     3   TGCA..CCGA             14              29              23               2              19


.. note:: Input files are automatically dectected from the input directory based on their extensions. Supported extensions are **.fa**, **.fa.gz**, **.fq**, **.fq.gz**, **.fasta**, **.fasta.gz**, **.fastq**, **.fastq.gz**, **.fna**, and **.fna.gz**. Individual IDs in the output table will be inferred from the input files names after removing the extension. For instance, a file named **individual_1.fastq.gz** will be attributed the ID **individual_1**. Make sure to use the same IDs when creating the population map.


.. _distrib-usage:

distrib
-------

Generate a table containing the distribution of markers between two groups of individuals defined in a :ref:`popmap-file`.

**Command**

::

    radsex distrib --markers-table markers_table_path --output-file output_file_path --popmap popmap_path [ --min-depth min_depth --groups group1,group2 --signif-threshold signif_threshold --disable-correction ]

**Options**

============================   ===========
Option                         Description
============================   ===========
``--markers-table, -t``        Path to a marker depths table generated by "process"
``--popmap, -p``               Path to a tabulated file specifying groups for all individuals
``--output-file, -o``          Path to the output file (distribution of markers between groups)
``--min-depth, -d``            Minimum depth to consider a marker present in an individual
``--groups, -G``               Names of the groups to compare if there are more than two groups in the popmap (--groups group1, group2)
``--signif-threshold, -S``     P-value threshold to consider a marker significantly associated with a phenotypic group
``--disable-correction, -C``   If set, Bonferroni correction will NOT be used when assessing significance of association with phenotypic group 
============================   ===========

**Output**

A :ref:`sex-distribution-file`:

::

    Males    Females    Markers       P     Signif      Bias
        0          1          7       1     False     -0.333
        0          2          3    0.39     False     -0.666
        0          3          1    0.10     False     -1.000
        1          0          6       1     False      0.333
        1          1          5       1     False      0.000
        1          2          1       1     False     -0.333
        1          3          2    0.39     False     -0.666
        2          0          3    0.39     False      0.666
        2          1          8       1     False      0.333
        2          2          4       1     False      0.000
        2          3          2       1     False     -0.333
        3          0          4    0.10     False      1.000
        3          1          7    0.39     False      0.666
        3          2          6       1     False      0.333
        3          3          9       1     False      0.000


.. _subset-usage:

subset
------

Extract a subset of markers from a markers depth table based on presence in individuals from each compared group.

**Command**

::

    radsex distrib --markers-table markers_table_path --output-file output_file_path --popmap popmap_path [ --min-depth min_depth --groups group1,group2 --signif-threshold signif_threshold --disable-correction --min-group1 min_group1 --min-group2 min_group2 --max-group1 max_group1 --max-group2 max_group2 --min-individuals min_individuals --max-individuals max_individuals --output-fasta ]

**Options**

============================   ===========
Option                         Description
============================   ===========
``--markers-table, -t``        Path to a marker depths table generated by "process"
``--popmap, -p``               Path to a tabulated file specifying groups for all individuals
``--output-file, -o``          Path to the output file (subset of markers)
``--min-depth, -d``            Minimum depth to consider a marker present in an individual
``--groups, -G``               Names of the groups to compare if there are more than two groups in the popmap (--groups group1, group2)
``--signif-threshold, -S``     P-value threshold to consider a marker significantly associated with a phenotypic group
``--disable-correction, -C``   If set, Bonferroni correction will NOT be used when assessing significance of association with phenotypic group
``--output-fasta, -a``         If set, markers are saved to a fasta file instead of a markers depth table
``--min-group1, -m``           Minimum number of individuals from the first group to retain a marker in the subset
``--min-group2, -n``           Minimum number of individuals from the second group to retain a marker in the subset
``--max-group1, -M``           Maximum number of individuals from the first group to retain a marker in the subset
``--max-group2, -N``           Maximum number of individuals from the second group to retain a marker in the subset
``--min-individuals, -i``      Minimum number of individuals to retain a marker in the subset
``--max-individuals, -I``      Maximum number of individuals to retain a marker in the subset
============================   ===========

**Output**

:ref:`markers-depths-table-file`:

::

    #Number of markers : 4
    id   sequence     individual_1    individual_2    individual_3    individual_4    individual_5
     0   TGCA..TATT              0              15              24              17              21
     1   TGCA..GACC             20              18               3              26               4
     2   TGCA..ATCG              2               1               5              16               0
     3   TGCA..CCGA             14              29              23               2              19

:ref:`fasta-file`:

::

    >4495827_F:0_M:21_p:1.14577e-07_mindepth:10
    TGCAG.....GGATGTGTATCCATA
    >4609394_F:0_M:23_p:1.08057e-08_mindepth:10
    TGCAG.....GGTTCCATCCCAAAT
    >4661391_F:0_M:26_p:1.92225e-10_mindepth:10
    TGCAG.....GTAGAGTGACCAGTT
    >5182569_F:0_M:22_p:3.62243e-08_mindepth:10
    TGCAG.....ACATGCTGTAAATGC
    >5625957_F:0_M:25_p:7.91633e-10_mindepth:10
    TGCAG.....CTGGAGAAGAGTAGG


.. _signif-usage:

signif
------

Find markers significantly associated with groups of individuals defined in a :ref:`popmap-file`.

**Command**

::

    radsex signif --markers-table markers_table_path --output-file output_file_path --popmap popmap_path [ --min-depth min_depth --groups group1,group2 --signif-threshold signif_threshold --disable-correction --output-fasta ]

**Options**

============================   ===========
Option                         Description
============================   ===========
``--markers-table, -t``        Path to a marker depths table generated by "process"
``--popmap, -p``               Path to a tabulated file specifying groups for all individuals
``--output-file, -o``          Path to the output file (distribution of markers between groups)
``--min-depth, -d``            Minimum depth to consider a marker present in an individual
``--groups, -G``               Names of the groups to compare if there are more than two groups in the popmap (--groups group1, group2)
``--signif-threshold, -S``     P-value threshold to consider a marker significantly associated with a phenotypic group
``--disable-correction, -C``   If set, Bonferroni correction will NOT be used when assessing significance of association with phenotypic group
``--output-fasta, -a``         If set, markers are saved to a fasta file instead of a markers depth table
============================   ===========

**Output**

:ref:`markers-depths-table-file`:

::

    #Number of markers : 4
    id   sequence     individual_1    individual_2    individual_3    individual_4    individual_5
     0   TGCA..TATT              0              15              24              17              21
     1   TGCA..GACC             20              18               3              26               4
     2   TGCA..ATCG              2               1               5              16               0
     3   TGCA..CCGA             14              29              23               2              19

:ref:`fasta-file`:

::

    >4495827_F:0_M:21_p:1.14577e-07_mindepth:10
    TGCAG.....GGATGTGTATCCATA
    >4609394_F:0_M:23_p:1.08057e-08_mindepth:10
    TGCAG.....GGTTCCATCCCAAAT
    >4661391_F:0_M:26_p:1.92225e-10_mindepth:10
    TGCAG.....GTAGAGTGACCAGTT
    >5182569_F:0_M:22_p:3.62243e-08_mindepth:10
    TGCAG.....ACATGCTGTAAATGC
    >5625957_F:0_M:25_p:7.91633e-10_mindepth:10
    TGCAG.....CTGGAGAAGAGTAGG


.. _map-usage:

map
---

**Command**

::

    radsex map --input-file input_file_path --output-file output_file_path --popmap-file popmap_file_path --genome-file genome_file_path [ --min-depth min_depth --min-quality min_quality --min-frequency min_frequency ]

The ``map`` command aligns all makers from a coverage table (obtained either from ``process``, ``subset``, or ``signif``) to a reference genome provided in fasta format. The output is a tabulated file where each line gives a marker ID, the contig where the marker mapped, the mapping position of the marker on this contig, the sex-bias of the marker (defined as M / Tm - F / Tf where M and F are the number of males and females in which the marker is present, and Tm and Tf are the total number of males and females in the population), the probability of association with sex for this marker (obtained with a chi-square test with Yate's correction for continuity), and the significativity of the association with sex after Bonferroni correction.

**Options**

=====================  ===========
Option                 Description
=====================  ===========
``--input-file``       Path to an coverage table obtained with ``process``, ``subset``, or ``signif``
``--output-file``      Path to the output file (in tsv format)
``--popmap-file``      Path to a popmap file indicating the sex of each individual
``--genome-file``      Path to a reference genome file in fasta format
``--min-depth``     Minimum coverage to consider a marker present in an individual (default: 1)
``--min-quality``      Minimum mapping quality, as defined in BWA, to consider a sequence properly mapped (default: 20)
``--min-frequency``    Minimum frequency in at least one sex for a sequence to be retained (default: 0.25)
=====================  ===========

**Sample output**

::

    Marker        Contig      Position      SexBias        P     Signif
    0             LG09        10052920            0        1     False
    1             LG45         4008419            0        1     False
    2             LG06        20521435            0        1     False
    3             LG24         7643946         0.13     0.44     False
    4             LG06        16975491            0        1     False
    5             LG27        16580048            0        1     False
    6             LG49         7206356         0.03        1     False
    7             LG30         5571989            0        1     False
    8             LG05        20094761            0        1     False
    9             LG14        20088495            0        1     False
    10            LG34        11566459        -0.04        1     False
    11            LG21        17338149            0        1     False
    12            LG05        14652417         0.13     0.55     False
    13            LG25        23851527         0.75    0.001     True


.. _freq-usage:

freq
----

**Command**

::

    radsex freq --input-file input_file_path --output-file output_file_path [ --min-depth min_depth ]

The ``freq`` command computes the distribution of markers in the population. The output is a tabulated file where the first column gives the number of individuals, and the second column gives the number of markers found in this number of individuals.

**Options**

=====================  ===========
Option                 Description
=====================  ===========
``--input-file``       Path to an coverage table obtained with ``process``
``--output-file``      Path to the output file (in tsv format)
``--min-depth``     Minimum coverage to consider a marker present in an individual (default: 1)
=====================  ===========

**Sample output**

::

    Frequency    Count
            1       15
            2       37
            3       48
            4       43
            5      124


.. _depth-usage:

depth
-----

**Command**

::

    radsex freq --input-file input_file_path --output-file output_file_path [ --min-depth min_depth ]

The ``freq`` command computes the distribution of markers in the population. The output is a tabulated file where the first column gives the number of individuals, and the second column gives the number of markers found in this number of individuals.

**Options**

=====================  ===========
Option                 Description
=====================  ===========
``--input-file``       Path to an coverage table obtained with ``process``
``--output-file``      Path to the output file (in tsv format)
``--min-depth``     Minimum coverage to consider a marker present in an individual (default: 1)
=====================  ===========

**Sample output**

::

    Frequency    Count
            1       15
            2       37
            3       48
            4       43
            5      124

