

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Getting started &mdash; RADSex 0.1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Example walkthrough" href="example.html" />
    <link rel="prev" title="RADSex documentation" href="index.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> RADSex
          

          
          </a>

          
            
            
              <div class="version">
                0.1
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Getting started</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#installation">Installation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#requirements">Requirements</a></li>
<li class="toctree-l3"><a class="reference internal" href="#install-release">Installation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#update-radsex">Update RADSex</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#before-starting">Before starting</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-radsex">Running RADSex</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#computing-the-markers-table">Computing the markers table</a></li>
<li class="toctree-l3"><a class="reference internal" href="#computing-the-distribution-of-markers-between-sexes">Computing the distribution of markers between sexes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#extracting-markers-significantly-associated-with-sex">Extracting markers significantly associated with sex</a></li>
<li class="toctree-l3"><a class="reference internal" href="#mapping-markers-to-a-reference-genome">Mapping markers to a reference genome</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="example.html">Example walkthrough</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage.html">RADSex usage details</a></li>
<li class="toctree-l1"><a class="reference internal" href="input_files.html">Input file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="output_files.html">Output file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="license.html">License</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">RADSex</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Getting started</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/getting_started.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="getting-started">
<h1>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
<div class="section" id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h2>
<div class="section" id="requirements">
<h3>Requirements<a class="headerlink" href="#requirements" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>A C++11 compliant compiler (GCC &gt;= 4.8.1, Clang &gt;= 3.3)</p></li>
<li><p>The zlib library (which should be installed on linux by default)</p></li>
</ul>
</div>
<div class="section" id="install-release">
<span id="id1"></span><h3>Installation<a class="headerlink" href="#install-release" title="Permalink to this headline">¶</a></h3>
<p>RADSex can be installed from one of the release packages [TODO: link], or the latest stable development version can be installed directly from the GitHub repository.</p>
<p><strong>1. Install the latest release</strong></p>
<p>RADSex is currently in beta and there is no available release yet. This section will be updated with the release of RADSex 1.0.</p>
<p><strong>2. Install from GitHub</strong></p>
<p>To install the latest stable version of RADSex from the GitHub repository, run the following commands:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">RomainFeron</span><span class="o">/</span><span class="n">RadSex</span><span class="o">.</span><span class="n">git</span>
<span class="n">cd</span> <span class="n">RadSex</span>
<span class="n">make</span>
</pre></div>
</div>
<p>The compiled <strong>radsex</strong> binary will be located in <strong>RadSex/bin/</strong>.</p>
</div>
<div class="section" id="update-radsex">
<h3>Update RADSex<a class="headerlink" href="#update-radsex" title="Permalink to this headline">¶</a></h3>
<p>To update RADSex, you can download the latest stable release and install it as described in the <a class="reference internal" href="#install-release"><span class="std std-ref">Installation</span></a> section.</p>
<p>If you installed RADSex from Github, run the following commands from the RadSex directory:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">pull</span>
<span class="n">make</span> <span class="n">rebuild</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="before-starting">
<h2>Before starting<a class="headerlink" href="#before-starting" title="Permalink to this headline">¶</a></h2>
<p>Before running the pipeline, you should prepare the following elements:</p>
<ul class="simple">
<li><p>A <strong>set of demultiplexed reads</strong>. The current version of RADSex does not implement demultiplexing. Raw sequencing reads can be demultiplexed using <a class="reference external" href="http://catchenlab.life.illinois.edu/stacks/comp/process_radtags.php">Stacks</a> or <a class="reference external" href="http://nbviewer.jupyter.org/gist/dereneaton/af9548ea0e94bff99aa0/pyRAD_v.3.0.ipynb#The-seven-steps-described">pyRAD</a>.</p></li>
<li><p>A <strong>population map</strong>: a tabulated file with individual ID as the first column and sex as the second column. It is important that the individual IDs in the popmap are the same as the names of the demultiplexed reads files (see the [popmap section](#population-map) for details).</p></li>
<li><p>If you want to map the sequences to a reference genome: a <strong>reference genome</strong> in fasta format.</p></li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When visualizing <cite>map</cite> results with <cite>radsex-vis</cite>, linkage groups / chromosomes are automatically inferred from scaffold names in the reference sequence if their name starts with <em>LG</em>, <em>CHR</em>, or <em>NC</em> (case unsensitive). If chromosomes are named differently in the reference genome, you should prepare a tabulated file with reference scaffold ID in the first column and corresponding chromosome name in the second column (see the [chromosomes names section](#chromosomes-names) for details).</p>
</div>
</div>
<div class="section" id="running-radsex">
<h2>Running RADSex<a class="headerlink" href="#running-radsex" title="Permalink to this headline">¶</a></h2>
<div class="section" id="computing-the-markers-table">
<span id="computing-cov-table"></span><h3>Computing the markers table<a class="headerlink" href="#computing-the-markers-table" title="Permalink to this headline">¶</a></h3>
<p>The first step of RADSex is to create a table of marker depth for the dataset using the <code class="docutils literal notranslate"><span class="pre">process</span></code> command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsex</span> <span class="n">process</span> <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="nb">dir</span> <span class="o">./</span><span class="n">samples</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">file</span> <span class="n">markers_table</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">16</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">depth</span> <span class="mi">1</span>
</pre></div>
</div>
<p>In this example, demultiplexed reads are stored in <strong>./samples</strong> and the markers table generated by <code class="docutils literal notranslate"><span class="pre">process</span></code> will be stored in <strong>markers_table.tsv</strong>. The parameter <code class="docutils literal notranslate"><span class="pre">--threads</span></code> specifies the number of threads to use, and <code class="docutils literal notranslate"><span class="pre">--min-depth</span></code> specifies the minimum depth to consider a marker present in an individual: markers which are not present with depth higher than this value in at least one individual will not be retained in the markers table.
It is advised to keep the minimum depth to 1 for this step, as it can be adjusted for each analysis later.</p>
<p>The resulting file <strong>markers_table.tsv</strong> is a table with N + 2 columns, where <em>N</em> is the number of individuals in the dataset :</p>
<ul class="simple">
<li><p><strong>ID</strong> : marker ID.</p></li>
<li><p><strong>Sequence</strong> : marker sequence.</p></li>
<li><p>For each individual, the depth of this marker.</p></li>
</ul>
</div>
<div class="section" id="computing-the-distribution-of-markers-between-sexes">
<h3>Computing the distribution of markers between sexes<a class="headerlink" href="#computing-the-distribution-of-markers-between-sexes" title="Permalink to this headline">¶</a></h3>
<p>After generating the markers table, the <code class="docutils literal notranslate"><span class="pre">distrib</span></code> command is used to compute the distribution of markers between sexes:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>radsex distrib --markers-table markers_table.tsv --output-file distribution.tsv --popmap popmap.tsv --min-depth 5``
</pre></div>
</div>
<p>In this example, the value of <code class="docutils literal notranslate"><span class="pre">--markers-table</span></code> is the table generated in the <a class="reference internal" href="#computing-cov-table"><span class="std std-ref">Computing the markers table</span></a> section, and the distribution of markers between sexes will be stored in <strong>distribution.tsv</strong>.
The sex of each individual in the population is given by <strong>popmap.tsv</strong> (see the [popmap section](#population-map) for details).
The minimum depth to consider a marker present in an individual is set to 5, meaning that markers with depth lower than 5 in an individual will not be considered present in this individual.</p>
<p>The resulting file <strong>distribution.tsv</strong> is a table with five columns:</p>
<ul class="simple">
<li><p><strong>Males</strong> : number of males in which a marker was present.</p></li>
<li><p><strong>Females</strong> : number of females in which a marker was present.</p></li>
<li><p><strong>Markers</strong> : number of markers present in the corresponding number of males and females.</p></li>
<li><p><strong>P</strong> : p-value of a chi-squared test for association with sex.</p></li>
<li><p><strong>Signif</strong> : significant association with sex (True / False).</p></li>
</ul>
<p>More details about the distribution file can be found in the [TODO SECTION].</p>
<p>This distribution can be visualized with the <code class="docutils literal notranslate"><span class="pre">plot_sex_distribution()</span></code> function of <a class="reference external" href="https://github.com/RomainFeron/RADSex-vis">RADSex-vis</a>, which generates a heatmap of markers with males on the x-axis and females on the y-axis.</p>
</div>
<div class="section" id="extracting-markers-significantly-associated-with-sex">
<h3>Extracting markers significantly associated with sex<a class="headerlink" href="#extracting-markers-significantly-associated-with-sex" title="Permalink to this headline">¶</a></h3>
<p>Markers significantly associated with sex can be obtained with the <code class="docutils literal notranslate"><span class="pre">signif</span></code> command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsex</span> <span class="n">signif</span> <span class="o">--</span><span class="n">markers</span><span class="o">-</span><span class="n">table</span> <span class="n">markers_table</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">file</span> <span class="n">markers</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">popmap</span> <span class="n">popmap</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">depth</span> <span class="mi">5</span> <span class="p">[</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="nb">format</span> <span class="n">fasta</span> <span class="p">]</span>
</pre></div>
</div>
<p>In this example, the value of <code class="docutils literal notranslate"><span class="pre">--markers-table</span></code> is the table generated in the <a class="reference internal" href="#computing-cov-table"><span class="std std-ref">Computing the markers table</span></a> section, and the markers significantly associated with sex are output in <strong>markers.tsv</strong>. The sex of each individual in the population is given by <strong>popmap.tsv</strong> (see the [popmap section](#population-map) for details), and the minimum depth to consider a marker present in an individual is set to 5, meaning that markers with depth lower than 5 in an individual will not be considered present in this individual.</p>
<p>By default, the <code class="docutils literal notranslate"><span class="pre">signif</span></code> function generates an output file in the same format as the markers table. However, the sequences can be exported to fasta using the <code class="docutils literal notranslate"><span class="pre">--output-fasta</span></code> parameter (see TODO SECTION).</p>
<p>The markers table generated by <code class="docutils literal notranslate"><span class="pre">signif</span></code> can be visualized with the <code class="docutils literal notranslate"><span class="pre">plot_coverage()</span></code> function of <a class="reference external" href="https://github.com/RomainFeron/RADSex-vis">RADSex-vis</a>, which generates a heatmap showing the depth of each marker in each individual.</p>
</div>
<div class="section" id="mapping-markers-to-a-reference-genome">
<h3>Mapping markers to a reference genome<a class="headerlink" href="#mapping-markers-to-a-reference-genome" title="Permalink to this headline">¶</a></h3>
<p>Markers can be aligned to a reference genome using the <code class="docutils literal notranslate"><span class="pre">map</span></code> command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">radsex</span> <span class="nb">map</span> <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="n">file</span> <span class="n">markers_table</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">file</span> <span class="n">mapping</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">popmap</span> <span class="n">popmap</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">genome</span><span class="o">-</span><span class="n">file</span> <span class="n">genome</span><span class="o">.</span><span class="n">fasta</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">quality</span> <span class="mi">20</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">frequency</span> <span class="mf">0.1</span> <span class="o">--</span><span class="nb">min</span><span class="o">-</span><span class="n">depth</span> <span class="mi">5</span>
</pre></div>
</div>
<p>In this example, the input file <code class="docutils literal notranslate"><span class="pre">--input-file</span></code> is the coverage table generated in the <a class="reference internal" href="#computing-cov-table"><span class="std std-ref">Computing the markers table</span></a> step, the mapping results will be stored in <strong>sequences.tsv</strong>, and the path to the reference genome file is given by <code class="docutils literal notranslate"><span class="pre">--genome-file</span></code>. The sex of each individual in the population is given by <strong>popmap.tsv</strong> (see the [popmap section](#population-map) for details), and the minimum coverage to consider a marker present in an individual is set to 5, meaning that markers with coverage lower than 5 in an individual will not be considered present in this individual. The parameter <code class="docutils literal notranslate"><span class="pre">--min-quality</span></code> specifies the minimum mapping quality (as defined in <a class="reference external" href="http://bio-bwa.sourceforge.net/bwa.shtml">BWA</a>) to consider a marker properly mapped, and is set to 20 in this example. The parameter <code class="docutils literal notranslate"><span class="pre">--min-frequency</span></code> specifies the minimum frequency of a marker in at least one sex; it is set to 0.1 here, meaning that only sequences present in at least 10% of individuals of one sex are retained for mapping.</p>
<p>The resulting file <code class="docutils literal notranslate"><span class="pre">mapping.tsv</span></code> is a table with five columns:</p>
<ul class="simple">
<li><p><strong>Sequence :</strong> ID of the mapped sequence.</p></li>
<li><p><strong>Contig :</strong> ID of the contig where the sequence mapped.</p></li>
<li><p><strong>Position :</strong> position of the mapped sequence on the contig.</p></li>
<li><p><strong>SexBias :</strong> sex-bias of the mapped sequence, defined as (Males / Total males ) - (Females / Total females), where <em>Males</em> and <em>Females</em> are the number of males and number of females in which the sequence is present, respectively, and <em>Total males</em> and <em>Total females</em> are the total number of males and females in the population, respectively.</p></li>
<li><p><strong>P :</strong> p-value of a chi-squared test for association with sex.</p></li>
<li><p><strong>Signif</strong> : significant association with sex (True / False).</p></li>
</ul>
<p>The mapping results generated by <code class="docutils literal notranslate"><span class="pre">map</span></code> can be visualized with the <code class="docutils literal notranslate"><span class="pre">plot_genome()</span></code> function of <a class="reference external" href="https://github.com/RomainFeron/RADSex-vis">RADSex-vis</a>, which generates a circular plot with the sex-bias and association with sex of each marker mapped on the genome.
Mapping results for a specific contig can be visualized with the <code class="docutils literal notranslate"><span class="pre">plot_scaffold()</span></code> function to show the same metrics for a single contig.</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="example.html" class="btn btn-neutral float-right" title="Example walkthrough" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="index.html" class="btn btn-neutral float-left" title="RADSex documentation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018-2020, Romain Feron

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>